source/js/ellipsis-util.js

```diff
@@ -0,0 +1,99 @@
+var ellipsis = require("html-ellipsis");
+var characterPerLine = require("./characters-per-line");
+
+var EllipsisUtil = {
+  /***
+   * @param {Object} lineStyles: Node with DOM elements that has a string
+   * @return {Boolean}
+   */
+  hasSpecificHeight: function(lineStyles) {
+    return lineStyles.height != "auto" && parseInt(lineStyles.height) != 0;
+  },
+
+  /***  
+   * @param {Node} element Node element.
+   * @return {Integer} Returns a font size.
+  * */
+  getFontSize: function(element) {
+    return parseInt(window.getComputedStyle(element).getPropertyValue("font-size"));
+  },
+
+  /***
+   * @summary Returns the width of the object 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} width
+   */
+  getWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+  
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      return parseInt(lineStyles.width);
+    }
+  },
+
+  /***
+   * @summary Returns the width of the object's parent 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} parent's width
+   */
+  getParentsWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+    
+    var lineStyles = window.getComputedStyle(textLine);
+          
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      var parentElement = textLine.parentElement;
+      var parentStyles = window.getComputedStyle(parentElement);
+
+      if(parentElement.className !== "multi-teaser-hotel-name") {
+        return parseInt(parentStyles.width);
+      }
+
+      var hotelRating = parentElement.querySelector(".mot_stars");
+      var hotelRatingWidth = window.getComputedStyle(hotelRating).width;
+
+      return parseInt(parentStyles.width) - parseInt(hotelRatingWidth) - parseInt(lineStyles.marginRight);
+    }
+  },
+};
+
+/*** 
+ * @summary Creates a EllipseableObject and pushes it into teaser. 
+ * @param {Node} element Node element.
+ * @param {Integer} numberOfLines One of the model's attributes.
+ * @param {Function} getStyles A self-made method to extract computed Styles.
+ */
+var EllipseableObject = function(element, getStylesFunction, numberOfLines){
```

Comment:
parameter "getStylesFunction" is not describe in the docBlock

Reply:
done